e2e/aks_model.go

```diff
@@ -428,3 +458,57 @@ func updateSubnet(ctx context.Context, cluster *armcontainerservice.ManagedClust
 	}
 	return nil
 }
+
+func createAzureResourceWithBicep(ctx context.Context, t *testing.T, params map[string]interface{}, resourceGroup, filename, deploymentName string) error {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("failed to get current working directory: %v", err)
+	}
+
+	bicepFileName := fmt.Sprintf("%s.bicep", filename)
+	bicepFilePath := filepath.Join(currentDir, "azure-resources", bicepFileName)
+	cmd := exec.Command("az", "bicep", "build", "--file", bicepFilePath)
```

Comment:
this is building the bicep file and then generating an ARM template?

Reply:
sad: https://github.com/Azure/bicep/issues/12893

Reply:
This will probably need extra movement to make it work in ADO/Github